ADAL/src/request/ADAuthenticationRequest+WebRequest.m

```diff
@@ -45,7 +46,8 @@ @implementation ADAuthenticationRequest (WebRequest)
 - (void)executeRequest:(NSDictionary *)request_data
             completion:(ADAuthenticationCallback)completionBlock
 {
-    NSString* urlString = [_context.authority stringByAppendingString:OAUTH2_TOKEN_SUFFIX];
+    NSString *authority = [NSString adIsStringNilOrBlank:_cloudAuthority] ? _context.authority : _cloudAuthority;
```

Comment:
It should be common for now, but we're also currently in process of discussing those scenarios with partners, figuring out how their own discovery works and defining the SSO story for sovereign aware scenarios. There might and might not be updates coming to the current caching logic in future.